### DISCUSSION TITLE
Skaffold Compatibility

### DISCUSSION THREAD

**Discussion by Exagram** [rancher-sandbox/rancher-desktop#1783] [Q&A]
Hi Rancher Community,

I have been currently using [Skaffold](https://skaffold.dev/) to automate development:
- Build Docker images on file changes
- Apply images to my local Minikube server

First problem: I try to run `skaffold dev` but it doesn't appear to work with `nerdctl/containerd`:

```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?. Docker build ran into internal error. Please retry.
If this keeps happening, please open an issue..
```

I fixed this by switching Rancher's default `containerd` option to `Docker Engine API`:

![image](https://user-images.githubusercontent.com/1795405/157742382-df4168a9-63d4-4b61-9d3a-b5da1031aa17.png)

Second problem: I ran `skaffold dev` but got back an error:
```
Build Failed. No push access to specified image repository. Trying running with `--default-repo` flag.
```

What image repository does Rancher/K3S use? Do I need to install extra software for this?

Thanks!

**Comment by gunamata** [comment]
For the first problem, skaffold doesn't work with nerdctl. So you are correct that you switched to `dockerd` to use `skaffold`

Second problem,

The default registry  is `https://index.docker.io/v1/` .  As you need to have push access to the image repository, you can either `docker login` (if you have docker hub account) before running `skaffold dev` or you can setup a local registry using `docker run -d -p 5000:5000 --restart=always --name registry registry:2` and then run `skaffold dev --default-repo=localhost:5000`.

Hope this helps!

**Reply by Exagram** [reply]
Hi @gunamata, I have succeeded in your steps above. How can I configure Rancher Kubernetes to use the local registry at `:5000`? I'm having difficulties finding a clear answer online.

**Reply by gunamata** [reply]
As Rancher Desktop uses k3s, you can use the instructions on this [page ](https://rancher.com/docs/k3s/latest/en/installation/private-registry/ )to configure private registry for your local kubernetes. 

I hope this helps!
